_config/mailer.yml

```diff
@@ -0,0 +1,34 @@
+---
+Name: mailer
+---
+SilverStripe\Core\Injector\Injector:
+  Symfony\Component\Mailer\MailerInterface:
+    class: Symfony\Component\Mailer\Mailer
+    constructor:
+      transport: '%$Symfony\Component\Mailer\Transport\TransportInterface'
+  Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer:
+    class: Symfony\Component\EventDispatcher\EventDispatcher
+    calls:
+      - [addSubscriber, ['%$SilverStripe\Control\Email\MailerSubscriber']]
+---
+Name: mailer-dsn-default-config
+---
+SilverStripe\Core\Injector\Injector:
+  Symfony\Component\Mailer\Transport\TransportInterface:
+    factory: Symfony\Component\Mailer\Transport
+    factory_method: fromDsn
+    constructor:
+      dsn: 'sendmail://default'
+      dispatcher: '%$Symfony\Component\EventDispatcher\EventDispatcherInterface.mailer'
```

Comment:
Have updated.  Now there's just a simple 'mailer' section with the default dsn and dispatcher.

Surprisingly, things work as expected if you have a project level config with `After: 'mailer'` i.e.
- sendmail://default is the dsn with no config
- the project level yml config is used if it has `After: 'mailer'`
- the .env level config is used if defined, it has precedence over the project level config.  I'm not entirely sure exactly this works since they both contain `After: 'mailer'. I would have expected the project config to override the framework env config, for some reason it's the other way around. Possibly the presence of `Only: envvarset:` somehow increases the priority of the env config

I've updated the developer-docs to use `After: 'mailer'`